Sources/PubNub/Subscription/ConnectionStatus.swift

```diff
@@ -60,6 +68,10 @@ public enum ConnectionStatus {
       return true
     case (.disconnectedUnexpectedly, _):
       return false
```

Comment:
It's covered in previous cases, and I took it from `master`. But your comment made me realize that this implementation is misleading. It's better to consider positive transitions only so the whole implementation would be more readable. I'm not sure yet if it's 100% correct, but I'm thinking about the following code snippet which I will test:

```
func canTransition(to state: ConnectionStatus) -> Bool {
    switch (self, state) {
    case (.connecting, .connected):
      return true
    case (.connecting, .disconnected):
      return true
    case (.connecting, .disconnectedUnexpectedly):
      return true
    case (.connecting, .connectionError):
      return true
    case (.reconnecting, .connected):
      return true
    case (.reconnecting, .disconnected):
      return true
    case (.reconnecting, .disconnectedUnexpectedly):
      return true
    case (.reconnecting, .connectionError):
      return true
    case (.disconnected, .connecting):
      return true
    case (.disconnectedUnexpectedly, .connecting):
      return true
    case (.connectionError, .connecting):
      return true
    default:
      return false
    }
}
```

Reply:
If we are connecting, then we probably can't transit to disconnected or disconnected unexpectedly because there are only two possible directions: connected or connection error - thoughts on only one status type.